lib/spi/addon-contracts.ts

```diff
@@ -25,5 +26,12 @@ export declare interface ClusterAddOn {
  * ClusterAddOn implementation may implement this interface in order to get post deployment hook point.
  */
 export declare interface ClusterPostDeploy {
-    postDeploy(clusterInfo: ClusterInfo, teams: Team[]): void;
-}
\ No newline at end of file
+    postDeploy(clusterInfo: types.ClusterInfo, teams: Team[]): void;
+}
+
+/** 
+ * Generic resource provider interface. 
+ **/
+export declare interface ResourceProvider<T extends IResource = IResource> {
```

Comment:
do you want this in the addon-contract file? Do we need a new resource-contract file?